scripts/backup-restore.sh

```diff
@@ -0,0 +1,414 @@
+    if [ -z "$snapshot_repo" ]; then
+      log "ERROR: Could not find OpenSearch snapshot repository."
+      exit 1
+    fi
+    log "Found snapshot repository: ${snapshot_repo}"
+
+    local snapshot_name
+    snapshot_name="manual-snapshot-$(date --utc +%Y%m%d%H%M%S)"
+    log "Taking snapshot: ${snapshot_name}"
+    curl -kL -u "${user}:${password}" -X PUT "${os_url}/_snapshot/${snapshot_repo}/${snapshot_name}" -H 'Content-Type: application/json' -d'
+      {
+        "indices": "*,-.opendistro_security",
+        "include_global_state": false
+      }
+      '
+
+    log "Waiting for snapshot to complete..."
+    local status=""
+    local retries=60 # 30 min timeout
+    while [ $retries -gt 0 ]; do
+      status=$(curl --insecure --location --silent --user "${user}:${password}" "${os_url}/_snapshot/${snapshot_repo}/${snapshot_name}" | jq -r '.snapshots[0].state')
+      log "Current snapshot status: ${status}"
+
+      if [ "$status" = "SUCCESS" ]; then
+        log "OpenSearch backup successfully completed."
+        return
+      elif [ "$status" = "FAILED" ] || [ "$status" = "PARTIAL" ]; then
+        log "ERROR: OpenSearch snapshot failed with status: ${status}"
+        exit 1
+      fi
+
+      sleep 30
+      retries=$((retries - 1))
+    done
+
+    log "ERROR: Timeout waiting for OpenSearch snapshot to complete."
+    exit 1
+    ;;
+  harbor)
+    log "Starting Harbor backup..."
+    local backup_job_name
+    backup_job_name="harbor-backup-manual-$(date --utc +%Y%m%d%H%M%S)"
+    log "Creating on-demand backup job: $backup_job_name"
+
+    "$CK8S_CMD" ops kubectl sc -n harbor create job "$backup_job_name" --from=cronjob/harbor-backup-cronjob
+
+    log "Waiting for backup job to complete..."
+    if ! "$CK8S_CMD" ops kubectl sc -n harbor wait --for=condition=complete "job/$backup_job_name" --timeout=30m; then
+      log "ERROR: Harbor backup job failed or timed out."
+      exit 1
+    fi
+
+    log "Backup complete. Deleting job: $backup_job_name"
+    "$CK8S_CMD" ops kubectl sc -n harbor delete job "$backup_job_name"
+
+    log "Harbor backup successfully completed."
+    ;;
+  rclone)
+    log "Starting Rclone backup..."
+
+    # Check if rclone-sync is enabled before continuing
+    log "Checking for rclone-sync cronjobs"
+    local cronjob_list
+    cronjob_list=$("$CK8S_CMD" ops kubectl sc -n rclone get cronjobs -lapp.kubernetes.io/instance=rclone-sync -oname)
+
+    if [ -z "$cronjob_list" ]; then
+      log "ERROR: No rclone-sync cronjobs found."
+      log "Please ensure rclone sync is enabled in your configuration and has been applied."
+      exit 1
+    fi
+    local total_jobs
+    total_jobs=$(echo "$cronjob_list" | wc -w)
+    log "Found ${total_jobs} cronjobs."
+
+    # Create jobs from cronjobs
+    for cronjob in $cronjob_list; do
+      "$CK8S_CMD" ops kubectl sc -n rclone create job --from "${cronjob}" "${cronjob/#cronjob.batch\//}"
+      local job_name=${cronjob/#cronjob.batch\//}
```

Comment:
Suggestion: Switch place on these and you don't need to do it twice

```suggestion
      local job_name=${cronjob/#cronjob.batch\//}
      "$CK8S_CMD" ops kubectl sc -n rclone create job --from "${cronjob}" "${job_name}"
```